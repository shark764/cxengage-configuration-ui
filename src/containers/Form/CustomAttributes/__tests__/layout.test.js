/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import CustomAttributesForm from '../layout';

describe('<CustomAttributesForm />', () => {
  it('renders create form', () => {
    const rendered = shallow(
      <CustomAttributesForm
        isSaving={false}
        inherited={false}
        handleSubmit={() => {}}
        userHasUpdatePermission={true}
        userHasCreatePermission={true}
        isCreatingNewAtrribute={true}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders update form', () => {
    const rendered = shallow(
      <CustomAttributesForm
        isSaving={false}
        inherited={false}
        handleSubmit={() => {}}
        userHasUpdatePermission={true}
        userHasCreatePermission={true}
        isCreatingNewAtrribute={false}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
