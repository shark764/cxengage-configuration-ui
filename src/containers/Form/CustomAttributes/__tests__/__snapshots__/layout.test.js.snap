// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomAttributesForm /> renders create form 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <Connect(DetailWrapper)
      data-automation="detailsSVG"
      open={true}
    >
      <Styled(DetailHeader)
        text="Details"
      />
      <InputField
        componentType="input"
        data-automation="attributeIdentifierInput"
        disabled={false}
        inputType="text"
        label="Attribute Identifier *"
        name="identifier"
      />
      <InputField
        componentType="input"
        data-automation="attributeNameInput"
        disabled={false}
        inputType="text"
        label="Attribute Name *"
        name="name"
      />
      <InputField
        componentType="input"
        data-automation="descriptionInput"
        disabled={false}
        inputType="text"
        label="Description"
        name="description"
      />
      <SelectField
        data-automation="dataTypeList"
        disabled={true}
        label="Data Type"
        name="dataType"
        options={
          Array [
            Object {
              "label": "Text",
              "value": "text",
            },
          ]
        }
        required={true}
      />
    </Connect(DetailWrapper)>
    <Connect(DetailWrapper)
      data-automation="availabilitySVG"
      open={true}
    >
      <Styled(DetailHeader)
        text="Availability"
      />
      <Styled(ToggleField)
        data-automation="realtimeToggle"
        disabled={false}
        label="Available for Realtime Reporting"
        labelMargin="0px"
        labelWidth="250px"
        name="realtime"
      />
      <Styled(ToggleField)
        data-automation="historicalToggle"
        disabled={false}
        label="Available for Historical Reporting"
        labelMargin="0px"
        labelWidth="250px"
        name="historical"
      />
    </Connect(DetailWrapper)>
  </styled.div>
</form>
`;

exports[`<CustomAttributesForm /> renders update form 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <Connect(DetailWrapper)
      data-automation="detailsSVG"
      open={true}
    >
      <Styled(DetailHeader)
        text="Details"
      />
      <InputField
        componentType="input"
        data-automation="attributeIdentifierInput"
        disabled={true}
        inputType="text"
        label="Attribute Identifier *"
        name="identifier"
      />
      <InputField
        componentType="input"
        data-automation="attributeNameInput"
        disabled={false}
        inputType="text"
        label="Attribute Name *"
        name="name"
      />
      <InputField
        componentType="input"
        data-automation="descriptionInput"
        disabled={false}
        inputType="text"
        label="Description"
        name="description"
      />
      <SelectField
        data-automation="dataTypeList"
        disabled={true}
        label="Data Type"
        name="dataType"
        options={
          Array [
            Object {
              "label": "Text",
              "value": "text",
            },
          ]
        }
        required={true}
      />
    </Connect(DetailWrapper)>
    <Connect(DetailWrapper)
      data-automation="availabilitySVG"
      open={true}
    >
      <Styled(DetailHeader)
        text="Availability"
      />
      <Styled(ToggleField)
        data-automation="realtimeToggle"
        disabled={false}
        label="Available for Realtime Reporting"
        labelMargin="0px"
        labelWidth="250px"
        name="realtime"
      />
      <Styled(ToggleField)
        data-automation="historicalToggle"
        disabled={false}
        label="Available for Historical Reporting"
        labelMargin="0px"
        labelWidth="250px"
        name="historical"
      />
    </Connect(DetailWrapper)>
  </styled.div>
</form>
`;
